pkg/backup/backupdest/backup_index_test.go

```diff
@@ -481,6 +483,308 @@ func TestIndexExists(t *testing.T) {
 	}
 }
 
+func TestListIndexes(t *testing.T) {
```

Comment:
In TestGetBackupIndexFileName, you could parse the filename with your new parser to make that test more of a roundtrip test. Also in that test, you could randomly assign some nice random  hour,min seconds to stress the parser a bit :D

Reply:
just commenting here, in case you missed it